osu.Game.Rulesets.Osu/Edit/Blueprints/HitCircles/Components/HitCircleOverlapMarker.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Utils;
+using osu.Game.Graphics;
+using osu.Game.Rulesets.Objects.Types;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.Skinning.Default;
+using osu.Game.Screens.Edit;
+using osu.Game.Skinning;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Osu.Edit.Blueprints.HitCircles.Components
+{
+    public class HitCircleOverlapMarker : BlueprintPiece<HitCircle>
+    {
+        /// <summary>
+        /// Hit objects are intentionally made to fade out at a constant slower rate than in gameplay.
+        /// This allows a mapper to gain better historical context and use recent hitobjects as reference / snap points.
+        /// </summary>
+        public const double FADE_OUT_EXTENSION = 700;
+
+        private readonly Circle circle;
+        private readonly RingPiece ring;
+
+        [Resolved]
+        private EditorClock editorClock { get; set; }
+
+        public HitCircleOverlapMarker()
+        {
+            Origin = Anchor.Centre;
+
+            Size = new Vector2(OsuHitObject.OBJECT_RADIUS * 2);
+
+            InternalChildren = new Drawable[]
+            {
+                circle = new Circle
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = Color4.White,
+                },
+                ring = new RingPiece
+                {
+                    BorderThickness = 4,
+                }
+            };
+        }
+
+        [Resolved]
+        private ISkinSource skin { get; set; }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            circle.BorderColour = colours.Yellow;
+        }
+
+        public override void UpdateFrom(HitCircle hitObject)
+        {
+            base.UpdateFrom(hitObject);
+
+            Scale = new Vector2(hitObject.Scale);
+
+            if ((hitObject is IHasComboInformation combo))
```

Comment:
nit: one pair of parens too many

```suggestion
            if (hitObject is IHasComboInformation combo)
```